aptos-move/block-executor/src/proptest_types/tests.rs

```diff
@@ -14,7 +14,10 @@ use crate::{
     },
     txn_commit_hook::NoOpTransactionCommitHook,
 };
-use aptos_types::executable::ExecutableTestType;
+use aptos_types::{
+    contract_event::{ContractEvent, ReadWriteEvent},
```

Comment:
The last comment re:traits, ExecutableTestType below is exception as it is used for transitional purposes everywhere, but typically we want to expose the trait type "ReadWriteEvent" here, but not the concrete type (ContractEvent), for example "TransactionWrite" vs "WriteOp" is a good example.

Long story short, what would be great to do here (in fact, in types.rs), is to introduce MockEvent type, let it implement ReadWriteEvent, and include it in the tests (and use instead of ContractEvent).
For now you can keep the implementation empty, but once we invoke the event patching logic in the executor, we will be able to use the Mock type to record and check things very easily (and will be independent of how the contract event actually evolves and inner implementation, V1, etc, it's actually happening at the moment).

Reply:
Added a `MockEvent` type now. Please let me know what you think.